core/src/calleeFunctions/index.ts

```diff
@@ -20,30 +27,95 @@ const allCalleeFunctions: Record<CalleeNames, CalleeFunctions> = {
     rETHCurveUniv3Callee,
 };
 
-export const getCalleeData = async function (
+export const getCalleesData = async function (
     network: string,
     collateralType: string,
     profitAddress: string
-): Promise<string> {
+): Promise<Record<string, string>> {
     const collateral = getCollateralConfigByType(collateralType);
-    const calleeFuctions = allCalleeFunctions[collateral.exchange.callee];
-    if (!calleeFuctions) {
+    const calleeFuctions = {} as Record<string, CalleeFunctions>;
+    Object.entries(collateral.exchanges).forEach(([key, value]) => {
+        calleeFuctions[key] = allCalleeFunctions[value.callee];
+    });
+    if (!Object.keys(calleeFuctions).length) {
         throw new Error(`Unsupported collateral type "${collateralType}"`);
     }
-    return await calleeFuctions.getCalleeData(network, collateral, profitAddress);
+    const calleesData = {} as Record<string, string>;
+    Object.entries(calleeFuctions).forEach(async ([key, value]) => {
+        calleesData[key] = await value.getCalleeData(network, collateral, profitAddress);
+    });
+    return calleesData;
 };
 
-const _getMarketPrice = async function (
+export const getMarketData = async function (
     network: string,
     collateralSymbol: string,
     amount: BigNumber = new BigNumber('1')
-): Promise<BigNumber> {
+): Promise<Record<string, MarketData>> {
     const collateral = getCollateralConfigBySymbol(collateralSymbol);
-    const calleeFuctions = allCalleeFunctions[collateral.exchange.callee];
-    if (!calleeFuctions) {
+    const calleeIdAndFuctions = {} as Record<string, CalleeFunctions>;
+    Object.keys(collateral.exchanges).forEach(id => {
+        calleeIdAndFuctions[id] = allCalleeFunctions[collateral.exchanges[id].callee];
+    });
```

Comment:
In general `{} as Record` and `calleeIdAndFuctions[id] = xxx` is not a preferred way of doing it in typescript. Every typecasting is bad and basically cancels data validation. Instead, you would want to create a loop over the keys that creates all necessary values and then returns them altogether as a complete object – this way you wouldn't need to cast anything.

It also goes along with the idea that modifying (eg reassigning values to the object keys in javascript is very tricky and you should avoid it) Why? Because objects are stored as references, so the references are passed between functions, not copies of the same object. If in some other function you will change a key, it will also be changed everywhere – including functions running in parallel. For this case, it's always better to avoid using `object[key] =` at all not validate every time what's going on

Reply:
The modern syntax is to use spread operator to recreate object `{ ...previousObject, newKey: newValue }` instead of using `Object.assign(Object.assign({}, previousObject), { newKey: newValue })` (which would do the same thing)